### PR TITLE
Fix endpoint muxer TODOs and make threading use vert.x.

### PR DESCRIPTION
**Pull request by master312**
⚠️⚠️⚠️ DO NOT MERGE: This PR is now deprecated. ⚠️⚠️⚠️

It has been superseded by https://github.com/ant-media/Ant-Media-Server/pull/8017.

Related Issue: https://github.com/ant-media/Ant-Media-Server/issues/7882

### CODE REVIEW

**Comment by dmtan90** [comment]
### 🤖 AI Code Review Summary
This is a significant and positive architectural shift from dedicated threads per muxer to a shared Vertx `WorkerExecutor` pool. This should greatly improve scalability under high load. The move to `executeBlocking` with a periodic timer is appropriate for this use case. My primary concerns are around native memory management (ensuring all `AVPacket`s in the queue are freed in all circumstances) and the potential CPU overhead of the 10ms periodic timer if the system scales to a large number of concurrent muxers. Please ensure the test suite is robust enough to cover the async nature of the new draining mechanism.

#### 📄 Inline File Comments:
- **src/main/java/io/antmedia/muxer/EndpointMuxer.java**: The `vertx.setPeriodic(10, ...)` timer runs every 10ms. This is quite aggressive and could lead to high CPU overhead if many muxers are active. Could this interval be adjusted or made configurable? Also, check if this timer is properly cancelled during all shutdown paths to prevent leaks.
- **src/main/java/io/antmedia/muxer/EndpointMuxer.java**: `LinkedBlockingQueue<AVPacket>`: Ensure all paths (especially error cases and teardown) correctly free these AVPacket instances to prevent native memory leaks. The current `freeQueuedPackets()` helper is a good start, but verify it covers all edge cases.
- **src/test/java/io/antmedia/test/MuxerUnitTest.java**: The test `testTeardownDrainsQueuedPackets` (renamed from `testShutdownWorkerQueueFullDrainsPacket`) likely needs to be carefully verified to ensure it properly mocks/handles the new Vertx-based worker pool lifecycle. Ensure that the test waits long enough for the async `drain()` task to complete before asserting the queue state.

File: src/main/server/conf/red5-common.xml [hunk 1]
```diff
@@ -140,6 +140,15 @@
    		<!--    * The default number of event loop threads to be used  = 2 * number of cores on the machine -->
     </bean>
```
Comment (burak-58) [inline src/main/server/conf/red5-common.xml:144]: I think we may use the available pool. If there is blocking error logs it gives us some information aabout the system.
